module/zfs/bpobj.c

```diff
@@ -263,22 +267,23 @@ propagate_space_reduction(bpobj_info_t *bpi, uint64_t freed,
 
 static int
 bpobj_iterate_blkptrs(bpobj_info_t *bpi, bpobj_itor_t func, void *arg,
-    dmu_tx_t *tx, boolean_t free)
+    int64_t start, dmu_tx_t *tx, boolean_t free)
```

Comment:
Looking back at the original internal review we had changed the `free` boolean argument here that naming in all the related functions from `free` to `delete` to make things more precise. Was there any reason why you didn't try to do that in this PR too? (e.g. is the change big enough that it could be in a separate PR? is the meaning of `free` more tightly coupled with other things here, than the original code in illumos? etc..)

Reply:
So the reason I originally replaced `free` with `delete` was because I wanted to use the `free` variable to indicate the status of the `blkptr`. I decided to use `bp_freed` for that instead and leave `free` as it was to minimize code change in this area. I don't have strong feelings on whether `free` or `delete` is more precise but I'm leaning towards sticking with the existing convention.

Reply:
That makes sense to me. Thank you for the clarification!